ADA/ada.qmd

```diff
@@ -19,57 +17,194 @@ It brings together:
+
+---
+
+## What this means for existing code
+
+### SQL scripts
+
+---
+
+#### Scripts currently used in SSMS
+
+If you currently use data and scripts stored in SSMS for your statistics publication pipeline, there will be small changes to your production pipelines that vary depending on your current process. 
+
+Your SQL code will need to be migrated over to Databricks to allow it to read from the Databricks SQL Warehouse and will require some minor changes to syntax as SSMS uses T-SQL and Databricks uses Spark SQL. The ADA team are currently developing a tool to allow you to easily convert T-SQL queries to Spark SQL syntax. 
+
+You can write new SQL code in the Databricks SQL Editor, which is very similar to SSMS. 
+
+---
+
+#### SQL scripts in repositories or embedded in R scripts
+
+These scripts do not need to be moved, but the syntax of the scripts or embedded code will need to be rewritten from T-SQL to Spark SQL. The only thing you will need to change is the connection source to allow the code to run. Rather than pointing to data held in current SQL databases, you will need to point towards the Databricks Delta Lake instead. 
+
+
+---
+
+#### Workflows
+
+---
+
+If you'd like to make use of automated workflows to run scheduled code, then your SQL code will need to be moved into Databricks notebooks, using Spark SQL syntax. 
+
+---
+
+### R scripts in RStudio
+
+---
+
+::: {.callout-important}
+If you have an existing RAP process and pipeline set up using RStudio and SSMS, there is no current expectation for you to migrate your existing R code or scripts from RStudio to Databricks notebooks. 
+:::
+
+Your R code can remain in RStudio and all existing R processes can still be run there. Code that points towards SQL scripts will need to be redirected to Databricks rather than SSMS. In order to use RStudio with a Databricks SQL warehouse, you will need to manually set up a connection. You can find more information about that on our [setup guidance page](/ADA/databricks_rstudio_sql_warehouse.html). 
 
 ---
 
-## Databricks code templates
+### Using notebooks
 
-See examples and code snippets in R, SQL and Python in these [Databricks code template notebooks](https://adb-6882499576863257.17.azuredatabricks.net/browse/folders/3070168888967919?o=6882499576863257#)
+---
+
+If you prefer using R in a notebook format and wish to use notebooks for your analysis, this is fine, but please consider the skill sets and knowledge levels in your team when doing so. For the purposes of RAP, analysis should be reproducible. If the rest of your team are unfamiliar with notebooks and will need to pick up your work at a later date, this could cause issues. The [Statistics Development Team](mailto:statistics.development@education.gov.uk) are happy to advise on this and provide guidance if you have any concerns. 
```

Comment:
I have a niggle in the back of my head that we're being too grumpy about new tools where notebooks are concerned. I agree they should not be the default but if there are any cases where we do think they can be helpful could we add them and add any warnings for how to use them effectively around those (like the team ability point is a good one)? 
From googling they seem good for data science and quick exploration, but not for heavy pipelines and should not be relied on for anything in production. 

I'd like to get a point that we're happy notebooks have a place for x , but not really ideal for y.

Reply:
The ad hoc analysis v regular pipeline is a really common RAP question - this might be a good way to pitch it? Maybe tweak the opening to:

'Notebooks are easy to use in databricks, and might be something you've not come across before. These can be helpful for ad hoc, exploratory or one off pieces of analysis to add additional documentation as you go, but please consider the skill sets and knowledge levels in your team when doing so. For regular analysis pipelines adopting RAP principles we'd usually advise against using notebooks and instead recommend a Git controlled repository of code scripts using the [standard repository template available in dfeR](https://dfe-analytical-services.github.io/dfeR/reference/create_project.html).

If the rest of your team... [continue as is already]'

What do you think @lauraselby / @jen-machin ?

Reply:
better, but can we add more to explain why we think this. 
My first response to "For regular analysis pipelines adopting RAP principles we'd usually advise against using notebooks and instead recommend a Git controlled repository of code scripts using the [standard repository template available in dfeR](https://dfe-analytical-services.github.io/dfeR/reference/create_project.html)." is but why? 
I don't think the team skills bit is enough of a reason, and coming from us could look a bit sad as we're there to help build skills (and we pushed so hard for everyone learning R over the last few years). There must be other reasons?

Reply:
Wary about saying "easy to use" but agree for the most part 😆 It sounds much friendlier - thanks!

Reply:
Added an extra sentence or two to link to ADA guidance on how to create notebooks

Reply:
I think we don't have the reasoning really clear yet that Laura's asked for, and probably want to go more explicit in tone and make it more like 'it's different in this x way and therefore we recommend y' to set notebooks in a really clear way. So imagine someone has never heard of a notebook before, what is our view on what they are and when they should be used.

Feels like one reason is inconsistency of code layout / structuring, which we currently dance around a bit in the opening.

How about stealing this table from the NHS R guidance on it and really pushing the version control and collaboration angle (as that is the key one for me, notebooks work well for individuals working alone, but less well for whole teams collaborating, I don't care that you can all edit simultaneously, thats is harmful to version controlling properly)

![image](https://github.com/user-attachments/assets/2c5bc8c6-430a-4dae-aa17-88e77077404f)

Also think this notebooks section doesn't belong here, no existing code is in notebooks, so I think we should be merging with the 'What is databricks?' section?

Reply:
ah Laura added that 2nd comment after I'd already made my edits! sorry 🤣

Reply:
For ease of reviewing I'm going to resolve this and open up new comments on the latest changed version